src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -5686,32 +5689,92 @@ void EmitAllAsciiContained()
                     return;
             }
 
-            // We determined that the character class may contain ASCII, so we
-            // output the lookup against the lookup table.
+            // We know that the whole class wasn't ASCII, and we don't know anything about the non-ASCII
+            // characters other than that some might be included, for example if the character class
+            // were [\w\d], so if ch >= 128, we need to fall back to calling CharInClass. For ASCII, we
+            // can use a lookup table, but if it's a known set of ASCII characters we can also use a helper.
 
-            // ch < 128 ? (bitVectorString[ch >> 4] & (1 << (ch & 0xF))) != 0 :
+            // ch < 128 ?
             Ldloc(tempLocal);
             Ldc(analysis.ContainsOnlyAscii ? analysis.UpperBoundExclusiveIfOnlyRanges : 128);
             Bge(comparisonLabel);
-            Ldstr(bitVectorString);
-            Ldloc(tempLocal);
-            Ldc(4);
-            Shr();
-            Call(s_stringGetCharsMethod);
-            Ldc(1);
-            Ldloc(tempLocal);
-            Ldc(15);
-            And();
-            Ldc(31);
-            And();
-            Shl();
-            And();
-            Ldc(0);
-            CgtUn();
+
+            // ASCII
+            switch (bitVectorString)
+            {
+                case "\0\0\0\u03ff\ufffe\u07ff\ufffe\u07ff":
```

Comment:
Same here can this be RegexCharClass.AsciiLetterOrDigit, etc.

Reply:
No, it can't. :)